pkg/summary/collector.go

```diff
@@ -97,19 +104,30 @@ func (tc *logCollector) CollectFailureUnit(name string, reason error) {
 func (tc *logCollector) CollectDuration(name string, t time.Duration) {
 	tc.mu.Lock()
 	defer tc.mu.Unlock()
-	tc.fields = append(tc.fields, zap.Duration(name, t))
+	val, ok := tc.durations[name]
+	if ok {
+		tc.durations[name] = val + t
+		return
+	}
+	tc.durations[name] = t
```

Comment:
This entire block can be simplified to

```go
tc.durations[name] += t
```

Demo: https://play.golang.org/p/5s8j5x7GSpl